[service.watchedlist] 1.2.3

SchapplM opened the pull request:
Please merge the latest fixes from https://github.com/SchapplM/xbmc-addon-service-watchedlist
- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

romanvm commented on service.watchedlist/lib/watchedlist/watchedlist.py:
```diff
@@ -568,6 +578,7 @@ def get_watched_xbmc(self, silent):
                           "id": 1})
                 if json_response.has_key('result') and json_response['result'] != None and json_response['result'].has_key('tvshows'):
```
.has_key() method is deprecated and should not be used. Correct syntax is: `if key in dict`

romanvm commented on service.watchedlist/lib/watchedlist/utils.py:
```diff
@@ -65,7 +65,9 @@ def setSetting(name,value):
-    return _Addon.getSetting(name)
+    # The returned value from getSetting is not unicode, only string
+    # This function returns unicode
+    return _Addon.getSetting(name).decode('utf-8')
```
Make sure that the setting was stored as utf-8 encoded string

SchapplM commented:
I implemented the requested changes (obfuscation, fix deprecated) and also let python code tools autopep8, pycodestyle and pylint search for code style violations and fixed most of them.
